src/main/frontend/handler/editor.cljs

```diff
@@ -3223,6 +3223,8 @@
    (when block-id
      (if-let [block (db-model/query-block-by-uuid block-id)]
        (or (db-model/has-children? block-id)
+           (some-> (:block/content block)
+                   (string/includes? "{{query"))
```

Comment:
> the query blocks are still expanded and collapsable

Can you still collapse the query's result when the current block (with the {{query inside) has no children? The intention is to enable collapsing/expanding of query results.

Reply:
For the childless block `{{query (and (priority B A))}}`, it looks like I can still collapse - https://www.loom.com/share/74f3f7df55504bc4b44b54b883111030. If there are some edge cases that require this code, is there a way we can pass the collapse state without hardcoding the macro name? It'd be nice to not have false positives like the one I showed above. I don't think there is without saving some of a block's ast but maybe you know

Reply:
>  it looks like I can still collapse

Yes, I haven't removed it.

> is there a way we can pass the collapse state without hardcoding the macro name?

Maybe a property like `logseq.block.type` to identify this block as a `query`?

> I don't think there is without saving some of a block's ast but maybe you know

Yes, I don't think we can use the block's AST since it's parsed during the rendering. How about modifying the property `logseq.block.type` (if needed) after the rendering? If the query is valid and its block is childless and non-collapsed, we can add the property so that it can be collapsable then.

Reply:
> Yes, I haven't removed it.

Ah. I'd removed it locally when I recorded the video

Reply:
Thanks to `:block/macros` we can use it to determine whether a block has a valid DSL query.
[11d300c](https://github.com/logseq/logseq/pull/8774/commits/11d300cfaa9f4c0e277544d8097d51de2aa4564d)

Reply:
Good idea. Pushed a tweak so that it only does this check on the query macro